cmd/sqlconfig/doc.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package sqlconfig
+
+/*
+Package sqlconfig defines the schema for the sqlconfig file. The sqlconfig file
+by default resides in the folder:
+
+	Windows: %USERPROFILE%\.sqlcmd\sqlconfig
+	*nix: ~/.sqlcmd/sqlconfig
+
+The sqlconfig contains Contexts.  A context is named (e.g. mssql2) and
+contains the Endpoint details (to connect to) and User details (to
+use for authentication with the endpoint.
+
+If there is more than one context defined, there is always a "currentcontext",
+the currentcontext can be changed using
+
+	sqlcmd config use-context CONTEXT_NAME
+
+# Example
+
+An example of the sqlconfig file looks like this:
+
+	apiversion: v1
+	endpoints:
+	- container:
+		id: 0e698e65e19d9c
+		image: mcr.microsoft.com/mssql/server:2022-latest
+	  endpoint:
```

Comment:
> endpoint: [](http://example.com/codeflow?start=2&length=10)

Let's pretend we will have scenarios involving non-TDS endpoints (eg PowerBI, postgres, SQL AS/RS/IS). Will adding those just be a matter of adding an optional "type" field to "endpoint" to identify it as needing a different driver? 

theoretically `sqlcmd query "some query"` could run any arbitrary query language if the query command were to route it to a package that knows how to connect to the non-TDS endpoint.

Reply:
I love the growth mind set here! :smile:

Reply:
Related to this, to be able the best connection strings from:

sqlcmd config connection-strings

Do we need to keep some metadata for the endpoint that tells us it is mssql-server, or mssql-edge, or azsqldb, or postgres! So we know how to generate a good connection string for it

Reply:
Added:

[https://github.com/microsoft/go-sqlcmd/issues/173](https://github.com/microsoft/go-sqlcmd/issues/173)

Should ~/.sqlcmd/sqlconfig keep metadata on what the endpoint is